CHANGELOG.md

```diff
@@ -8,6 +8,107 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+
+  *In cases where the index is already applied, this update have no effect*.
+
+  **Please Note**: In some cases where ingest is occurring at high volume levels and/or there are
+  record volumes in the files table of > 150M files records, the migration may
+  fail on deployment due to timing required to both acquire the table state needed for the
+  migration and time to create the index given the resources available.
+
+  For reference a rx.5 large Aurora/RDS database
+  with no activity took roughly 6 minutes to create the index for a 300M file
+  table with no ingest, however timed out when the same migration was attempted
+  in production with possible activity on the table.
+
+  If you believe you are subject to the above consideration, you may opt to
+  manually create the `files` table index *prior* to deploying this version of
+  Core with the following procedure:
+
+  -----
+
+  - Verify you do not have the index:
+
+  ```text
+  select * from pg_indexes where tablename = 'files';
+
+   schemaname | tablename |        indexname        | tablespace |                                       indexdef
+  ------------+-----------+-------------------------+------------+---------------------------------------------------------------------------------------
+   public     | files     | files_pkey              |            | CREATE UNIQUE INDEX files_pkey ON public.files USING btree (cumulus_id)
+   public     | files     | files_bucket_key_unique |            | CREATE UNIQUE INDEX files_bucket_key_unique ON public.files USING btree (bucket, key)
+  ```
+
+  In this instance you should not see an `indexname` row with
+  `files_granules_cumulus_id_index` as the value.     If you *do*, you should be
+  clear to proceed with the installation.
+  - Quiesce ingest
+
+  Stop all ingest operations in Cumulus Core according to your operational
+  procedures.    You should validate that it appears there are no active queries that
+  appear to be inserting granules/files into the database as a secondary method
+  of evaluating the database system state:
+
+  ```text
+  select pid, query, state, wait_event_type, wait_event from pg_stat_activity where state = 'active';
+  ```
+
+  If query rows are returned with a `query` value that involves the files table,
+  make sure ingest is halted and no other granule-update activity is running on
+  the system.
+
+  Note: In rare instances if there are hung queries that are unable to resolve, it may be necessary to
+  manually use psql `Server Signaling Functions` `pg_cancel_backend` and/or
+  `pg_terminate_backend` if the migration will not complete in the next step.
+
+  - Create the Index
+
+  Run the following query to create the index.    Depending on the situation
+  this may take many minutes to complete, and you will note your CPU load and
+  disk I/O rates increase on your cluster:
+
+  ```text
+  CREATE INDEX files_granule_cumulus_id_index ON files (granule_cumulus_id);
```

Comment:
Is there any value in `IF NOT EXISTS` if they're running it manually? Maybe if they're running this by mistake and have already added the index?

Reply:
I may have just missed this point in our convo 🤔

Reply:
If it *does* exist, it does something like: 

---
jk_tf4_db=> CREATE INDEX files_granule_cumulus_id_index ON files (granule_cumulus_id)                                                                                                                                     ;
ERROR:  relation "files_granule_cumulus_id_index" already exists
---

Which in the case of manual creation is probably.... useful?